### PR TITLE
Mobile: add tags filter to note tags dialog

### PR DESCRIPTION
**Pull request by shinglyu**
This PR adds a tag filter (the "Filter tags" input box in the screen recording below) for mobile app's note tag dialog. It helps the user search for a particular tag in the tags list. 

https://user-images.githubusercontent.com/3250983/156068396-e6b74860-78ec-4411-9f92-42f785899c43.mp4

### CODE REVIEW

File: packages/app-mobile/components/screens/NoteTagsDialog.js [hunk 1]
```diff
@@ -140,16 +145,16 @@ class NoteTagsDialogComponent extends React.Component {
 				fontSize: 20,
 				color: theme.color,
 			},
-			newTagBox: {
+			tagBox: {
 				flexDirection: 'row',
 				alignItems: 'center',
-				paddingLeft: theme.marginLeft,
-				paddingRight: theme.marginRight,
+				paddingLeft: 10,
```
Comment (shinglyu) [inline packages/app-mobile/components/screens/NoteTagsDialog.js:151]: Trying to make the left edge of the label aligns with the tags's left edge so it's more visually pleasing. But I'm fine with leaving this out.

File: packages/app-mobile/components/screens/NoteTagsDialog.js [hunk 2]
```diff
@@ -170,10 +175,23 @@ class NoteTagsDialogComponent extends React.Component {
 						onChangeText={value => {
 							this.setState({ newTags: value });
 						}}
-						style={this.styles().newTagBoxInput}
+						style={this.styles().tagBoxInput}
+						placeholder="e.g. foo,bar"
```
Comment (shinglyu) [inline packages/app-mobile/components/screens/NoteTagsDialog.js:179]: I added a placeholder here to help user understand that they can actually input a comma-seperated list of tags, which is not very self explanatory. But I'm OK to keep it as-is.

Comment (laurent22) [reply packages/app-mobile/components/screens/NoteTagsDialog.js:179]: Ok I think that helps, but "foo,bar" I think is hard to translate. Could you make it instead `_('tag1, tag2...')`?

**Comment by shinglyu** [comment]
Fixed! Thank you for your feedback @laurent22

**Comment by laurent22** [comment]
Great, thanks a lot for adding this useful feature @shinglyu!

**Pull request merged**